apps/autoscheduler/Featurization.cpp

```diff
@@ -0,0 +1,22 @@
+#include "Featurization.h"
+#include "FunctionDAG.h"
+
+namespace Halide {
+void compute_pipeline_featurization(const Pipeline &pipeline, const Target& tgt, const MachineParams &params, std::unordered_map<Stage, PipelineFeatures, StageHasher> *features) {
```

Comment:
I'm not sure this interface makes sense. You can get the pipeline features this way with a temporary internal FunctionDAG, but you also need a FunctionDAG to compute schedule-specific features, and the DAG is expensive to construct - it's where all the front-loaded analysis of the pipeline is done (e.g. bounds of things).